File: test/src/main/java/demo/lucius/test/MainActivity.java
package demo.lucius.test;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import dagger.android.support.DaggerAppCompatActivity;

import javax.inject.Inject;

public class MainActivity extends DaggerAppCompatActivity {

    @Inject
    LoginNavigation navigation;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: app/src/main/java/demo/lucius/androidprojectkt/MainActivity.kt
package demo.lucius.androidprojectkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import demo.lucius.androidprojectkt.util.autoCleared
import javax.inject.Inject
import kotlin.jvm.functions.FunctionN

class MainActivity : AppCompatActivity(),HasSupportFragmentInjector {


    var list by autoCleared<MutableList<String>>()

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    override fun supportFragmentInjector()= dispatchingAndroidInjector
}
